batch/batch/worker/worker.py

```diff
@@ -1502,31 +1505,46 @@ async def run(self):
                 if not user_error(e):
                     log.exception(f'while running {self}')
 
-                self.state = 'error'
-                self.error = traceback.format_exc()
+                if isinstance(e, JobDeletedError):
+                    self.state = 'cancelled'
```

Comment:
The changes to JobDeletedError have nothing to do with fixing status, right? That's just a change that will reduce exceptional logs, right?

Reply:
It will make cancelled jobs have non-error statuses and be marked cancelled instead. It's a bit strange to cancel a running Job (example, infinite loop) and then have the UI page show it as an error with no logs or output.